portal-impl/src/com/liferay/portal/repository/registry/RepositoryClassDefinitionCatalogImpl.java

```diff
@@ -111,6 +113,15 @@ public RepositoryClassDefinition getRepositoryClassDefinition(
 
 	@Override
 	public void invalidate() {
+		if (DBPartition.isPartitionEnabled() &&
```

Comment:
@marianoalvarosaiz @achaparro this change seems not covered by the previous integration test, please add coverage

Reply:
@tinatian can you point out what exactly is not covered by the test? The test class requires DB Partition enabled to run and the CompanyThreadLocal is always set different than SYSTEM in those cases.